.github/workflows/draft.yml

```diff
@@ -0,0 +1,60 @@
+# The present workflow was made based on the following references:
+# - https://evilmartians.com/chronicles/build-images-on-github-actions-with-docker-layer-caching
+# - https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md
+# - https://github.com/commercialhaskell/stack/blob/master/.github/workflows/integration-tests.yml
+---
+name: Draft
+
+on:
+  create:
+    tags:
+      - v*
+
+concurrency:
+  group: draft-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Cache Docker layers
+        uses: actions/cache@v3
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: ${{ runner.os }}-buildx-
+      - name: Build Docker image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          load: true
+          tags: ${{ github.repository }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache-new,mode=max
+      - name: Copy static binary
+        run: |
+          docker run \
+            --entrypoint cp \
+            --volume $PWD/bin:/root/bin \
+            ${{ github.repository }} \
+            /usr/local/bin/hap \
+            /root/bin/hap-${{ github.ref_name }}-linux-x86_64-bin
+      - name: Create draft release
+        uses: softprops/action-gh-release@v1
+        with:
+          files: |
+            bin/hap-*
+            LICENSE
```

Comment:
I was thinking that we could add some more things to the release. For example, many projects include a tarball of the projects source code at the point in which the commit tag was created. I think that would be easy to add (the one created with `cabal sdist` would work I think) and could be nice to have. Maybe we could address this in some other PR though.

I'm sure there was something else I had in mind some weeks ago but I can't remember right now.

Reply:
@DavidMazarro adding the output of the `cabal sdist` command as well as uploading a tarball with the source code makes sense to me, however, I think we could add that later as you suggested taking into account this PR is already quite large :)